MarkBind/markbind
# Implement global disable intralink validation

**What is the purpose of this pull request?**


Resolves #1418 requirement of disabling global intralink validation. 



**Overview of changes:**
Allow users to disable link validation globally by adding a field `validateIntralink` in their `site.json`. Link validation is enabled by default. 

This boolean value will be passed in a top-down manner to the link processing part, where we can use this flag to determine whether to validate the intralink or not. 

**Anything you'd like to highlight / discuss:**
If there is anything I missed out. 

**Testing instructions:**
Indicate `validateIntralink` in your site.json with the value set to **false** and you should not see any warnings for potential invalid links. 

Set the value to true or remove the field entirely, and you should see warnings for potential invalid links. 



**Proposed commit message: (wrap lines at 72 characters)**
Implement global disable intralink validation 

Users do not have a way to disable intralink validation globally. 

The feature is enabled by default but they may not want this feature to be present. 

Let's create a way for them to disable the intralink validation entirely. 



---

**Checklist:** :ballot_box_with_check:
